backend/service/k8s/lightweightinformer_test.go

```diff
@@ -0,0 +1,68 @@
+package k8s
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/cache"
+	fakecontroller "k8s.io/client-go/tools/cache/testing"
+)
+
+func TestLightweightInformer(t *testing.T) {
+	cs := fake.NewSimpleClientset()
+	csm := &managerImpl{
+		clientsets: map[string]*ctxClientsetImpl{"foo": &ctxClientsetImpl{
+			Interface: cs,
+			namespace: "default",
+			cluster:   "core-testing",
+		}},
+	}
+
+	addActionChan := make(chan int)
+	updateActionChan := make(chan int)
+	deleteActionChan := make(chan int)
+
+	informerHandlers := cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			addActionChan <- 1
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			updateActionChan <- 1
+		},
+		DeleteFunc: func(obj interface{}) {
+			deleteActionChan <- 1
+		},
+	}
+
+	// This fake controller is a fake source for listing and watching resources
+	fc := fakecontroller.NewFakeControllerSource()
+
+	stop := make(chan struct{})
+	defer close(stop)
+
+	podInformer := NewLightweightInformer(
+		csm.clientsets["foo"],
+		fc,
+		&v1.Pod{},
+		time.Minute,
+		informerHandlers,
+	)
+
+	go podInformer.Run(stop)
+
+	// We sleep below to give some time for the informer to get the event and hit a handler
+	fc.Add(&v1.Pod{})
+	time.Sleep(time.Millisecond * 50)
```

Comment:
Looking at this again, I probably dont need the sleeps at all since I wait for the channel to return something anyway